uaclient/entitlements/tests/test_esm.py

```diff
@@ -204,6 +235,8 @@ def test_disable_removes_apt_config(
         m_platform_info,
         m_remove_auth_apt_repo,
         m_remove_apt_list_files,
+        m_remove_repo_from_apt_auth_file,
```

Comment:
Ok, mocks are getting a bit out of control and maybe this unit test is trying to do too much. I'll mark WIP and let @OddBloke decide whether we need to make this more testable or make the unit test more concise now that we have quite distinct behaviors between esm-infra and esm-apps.

Reply:
Given the enormous if/else in here, I think splitting into separate tests makes sense. It's worth considering, however, that there may be _more_ than two tests here: there may be some common behaviour we're testing in this test which can continue to be shared between the two.

Reply:
now that test_repo.py covers most of setup_apt_config and remove_apt_config, we don't need to explicitly test that interaction within esm tests. I've updated unit tests and this is ready for re-review.